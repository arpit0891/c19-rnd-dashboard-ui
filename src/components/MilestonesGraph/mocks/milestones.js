export const milestones = [
  // {
  //   name: 'Optimistic',
  //   values: [
  //     {
  //       value: '15%',
  //       start: new Date('2020/02/10'),
  //       end: new Date('2020/02/20'),
  //     },
  //     {
  //       value: '20%',
  //       start: new Date('2020/03/01'),
  //       end: new Date('2020/03/15'),
  //     },
  //     {
  //       value: '15%',
  //       start: new Date('2020/03/17'),
  //       end: new Date('2020/03/28'),
  //     },
  //     {
  //       value: '5%',
  //       start: new Date('2020/04/04'),
  //       end: new Date('2020/04/23'),
  //     },
  //     {
  //       value: '10%',
  //       start: new Date('2020/04/25'),
  //       end: new Date('2020/05/05'),
  //     },
  //   ],
  // },
  // {
  //   name: 'Pesimistic',
  //   values: [
  //     {
  //       value: '20%',
  //       start: new Date('2020/02/21'),
  //       end: new Date('2020/03/10'),
  //     },
  //     {
  //       value: '25%',
  //       start: new Date('2020/03/23'),
  //       end: new Date('2020/04/25'),
  //     },
  //     {
  //       value: '30%',
  //       start: new Date('2020/04/26'),
  //       end: new Date('2020/05/04'),
  //     },
  //     {
  //       value: '10%',
  //       start: new Date('2020/05/20'),
  //       end: new Date('2020/06/15'),
  //     },
  //     {
  //       value: '15%',
  //       start: new Date('2020/06/18'),
  //       end: new Date('2020/06/30'),
  //     },
  //   ],
  // },
  {
    name: 'Actual',
    values: [
      {
        value: '16%',
        className: 'clinical_batch',
        start: new Date('2020/03/01'),
        end: new Date('2020/03/15'),
      },
      {
        value: '22%',
        className: 'phase_1',
        start: new Date('2020/03/01'),
        end: new Date('2020/03/15'),
      },
      {
        value: '19%',
        className: 'lead_selection',
        start: new Date('2020/03/01'),
        end: new Date('2020/03/15'),
      },
    ],
  },
]
