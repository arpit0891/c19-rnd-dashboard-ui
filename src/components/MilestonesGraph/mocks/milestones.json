[
  {
    "name": "Actual",
    "values": [
      {
        "className": "discovery",
        "end": "2020-01-13T00:00:00.000Z",
        "start": "2020-01-11T00:00:00.000Z",
        "value": "1%"
      },
      {
        "className": "lead_selection",
        "end": "2020-02-07T00:00:00.000Z",
        "start": "2020-01-13T00:00:00.000Z",
        "value": "20.86%"
      },
      {
        "className": "clinical_batch",
        "end": "2020-03-04T00:00:00.000Z",
        "start": "2020-02-07T00:00:00.000Z",
        "value": "21.69%"
      },
      {
        "className": "ind",
        "end": "2020-03-16T00:00:00.000Z",
        "start": "2020-03-04T00:00:00.000Z",
        "value": "10.01%"
      },
      {
        "className": "phase_1",
        "end": "2020-05-09T20:47:50.756Z",
        "start": "2020-03-16T00:00:00.000Z",
        "value": "45.77%"
      }
    ]
  }
]
