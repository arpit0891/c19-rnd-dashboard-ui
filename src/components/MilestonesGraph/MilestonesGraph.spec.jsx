import React from 'react'
import { shallow } from 'enzyme'
import {
  MilestonesGraph,
  getAllDatesFromMilestones,
  getEarliestDate,
  getLatestDate,
  getTotalMilestoneDurationInDays,
} from './MilestonesGraph'
import milestones from './mocks/milestones.json'

it('should render all milestones ', () => {
  const numberOfMilestones = milestones.length
  const wrapper = shallow(<MilestonesGraph milestones={milestones} />)
  expect(wrapper.find('[data-test-id="bar"]')).toHaveLength(numberOfMilestones)
  expect(wrapper.find('[data-test-id="label"]')).toHaveLength(
    numberOfMilestones
  )
})

it('should render start and end dates', () => {
  const wrapper = shallow(<MilestonesGraph milestones={milestones} />)
  const xAxis = wrapper.find('[data-test-id="x-axis"]')
  expect(
    xAxis.find('[data-test-id="start-date"]').prop('data-test-value')
  ).toEqual('2020-01-11T00:00:00.000Z')
  expect(
    xAxis.find('[data-test-id="end-date"]').prop('data-test-value')
  ).toEqual('2020-05-09T20:47:50.756Z')
})

const expectedDates = [
  {
    start: new Date('2020/02/10'),
    end: new Date('2020/02/20'),
  },
  {
    start: new Date('2020/03/01'),
    end: new Date('2020/03/15'),
  },
  {
    start: new Date('2020/03/17'),
    end: new Date('2020/03/28'),
  },
  {
    start: new Date('2020/04/04'),
    end: new Date('2020/04/23'),
  },
  {
    start: new Date('2020/04/25'),
    end: new Date('2020/05/05'),
  },
  {
    start: new Date('2020/02/21'),
    end: new Date('2020/03/10'),
  },
  {
    start: new Date('2020/03/23'),
    end: new Date('2020/04/25'),
  },
  {
    start: new Date('2020/04/26'),
    end: new Date('2020/05/04'),
  },
  {
    start: new Date('2020/05/20'),
    end: new Date('2020/06/15'),
  },
  {
    start: new Date('2020/06/18'),
    end: new Date('2020/06/30'),
  },
  {
    start: new Date('2020/03/01'),
    end: new Date('2020/03/15'),
  },
  {
    start: new Date('2020/03/01'),
    end: new Date('2020/03/15'),
  },
  {
    start: new Date('2020/03/01'),
    end: new Date('2020/03/15'),
  },
]

it('should get all the dates from the milestones', () => {
  expect(getAllDatesFromMilestones(milestones)).toMatchSnapshot()
})

it('should return the earliest date', () => {
  const dates = expectedDates.map(({ start }) => start)
  expect(getEarliestDate(dates)).toEqual(new Date('2020/02/10'))
})

it('should return the latest date', () => {
  const dates = expectedDates.map(({ end }) => end)
  expect(getLatestDate(dates)).toEqual(new Date('2020/06/30'))
})

// eslint-disable-next-line jest/no-disabled-tests
it.skip('should return the duration of the milestone in days', () => {
  const milestone = milestones[0]
  expect(getTotalMilestoneDurationInDays(milestone)).toEqual(85)
})
