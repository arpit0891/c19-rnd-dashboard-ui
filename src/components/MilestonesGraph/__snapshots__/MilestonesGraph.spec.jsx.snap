// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should get all the dates from the milestones 1`] = `
Array [
  Object {
    "end": "2020-01-13T00:00:00.000Z",
    "start": "2020-01-11T00:00:00.000Z",
  },
  Object {
    "end": "2020-02-07T00:00:00.000Z",
    "start": "2020-01-13T00:00:00.000Z",
  },
  Object {
    "end": "2020-03-04T00:00:00.000Z",
    "start": "2020-02-07T00:00:00.000Z",
  },
  Object {
    "end": "2020-03-16T00:00:00.000Z",
    "start": "2020-03-04T00:00:00.000Z",
  },
  Object {
    "end": "2020-05-09T20:47:50.756Z",
    "start": "2020-03-16T00:00:00.000Z",
  },
]
`;
